### PR TITLE
feat: allow to init remote access plugin with non-standard user/group

### PR DESCRIPTION
**Pull request by Ruadhri17**
## Proposed changes
This PR add two new flags to `c8y-remote-access-plugin`: `--user` and `--group` that can be used only with `--init` flag. Plugin will now create directory with custom permissions provided by user or change the permissions if file/directory already exists.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
* #2886
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by albinsuresh — DISMISSED** [review]
LGTM.

File: plugins/c8y_remote_access_plugin/src/input.rs [hunk 1]
```diff
@@ -58,6 +58,14 @@ pub struct C8yRemoteAccessPluginOpt {
     /// taking the SmartREST input as an argument
     // Use "-" to read the value from stdin.
     child: Option<String>,
+
+    /// The user who will own the directories created
```
Comment (didier-wenzek) [inline plugins/c8y_remote_access_plugin/src/input.rs:62]:
```suggestion
    /// The user who will own the directories created by --init
```

File: plugins/c8y_remote_access_plugin/src/input.rs [hunk 2]
```diff
@@ -58,6 +58,14 @@ pub struct C8yRemoteAccessPluginOpt {
...
+    #[arg(long, requires("init"), default_value = "tedge")]
+    user: Option<String>,
+
+    /// The group who will own the directories created
```
Comment (didier-wenzek) [inline plugins/c8y_remote_access_plugin/src/input.rs:66]:
```suggestion
    /// The group who will own the directories created by --init
```

File: plugins/c8y_remote_access_plugin/src/lib.rs [hunk 3]
```diff
@@ -40,59 +41,73 @@ pub async fn run(opt: C8yRemoteAccessPluginOpt) -> miette::Result<()> {
...
+            .context("Changing permissions of supported operations")
+    } else {
+        create_file_with_user_group(
+            supported_operation_path,
+            user,
+            group,
+            0o644,
+            Some(
+                r#"[exec]
+    command = "c8y-remote-access-plugin"
```
Comment (didier-wenzek) [inline plugins/c8y_remote_access_plugin/src/lib.rs:102]: The system test "Install/uninstall c8y-remote-access-plugin" is failing for this extra white space ;-), but better to fix the test removing the useless constraint on the `command` keyword having to start the line. 

```suggestion
command = "c8y-remote-access-plugin"
```

**Review by didier-wenzek — APPROVED** [review]
Approved: the system test is failing for a typo.

File: tests/RobotFramework/tests/cumulocity/remote-access/test_remote_access.robot [hunk 4]
```diff
@@ -16,14 +16,24 @@ Install/uninstall c8y-remote-access-plugin
...
     ...    timeout=5
     Should Match Regexp    ${executable}    pattern=^"c8y-remote-access-plugin\\b
 
     Execute Command    dpkg -r c8y-remote-access-plugin
     File Should Not Exist    /etc/tedge/operations/c8y/c8y_RemoteAccessConnect
 
+Init c8y-remote-access-plugin with the custom user and group
+    Device Should Have Installed Software    c8y-remote-access-plugin
+
+    Execute Command    sudo c8y-remote-access-plugin --init --user petertest --group petertest
```
Comment (rina23q) [inline tests/RobotFramework/tests/cumulocity/remote-access/test_remote_access.robot:30]: Minor suggestion. Using different values for --user and --group will potentially detect more bugs.

```suggestion
    Execute Command    sudo c8y-remote-access-plugin --init --user peter --group test
```

**Review by rina23q — APPROVED** [review]
Minor suggestion

File: tests/RobotFramework/tests/cumulocity/remote-access/test_remote_access.robot (hunk 4 above)
Comment (Ruadhri17) [reply tests/RobotFramework/tests/cumulocity/remote-access/test_remote_access.robot:30]: In this case I used the user/group that are present in the test container, so I would need to add new one to test different user and group. Besides, this test is just another variation of what we do with `tedge init` and I think it is sufficient for change that was implemented.

Comment (rina23q) [reply tests/RobotFramework/tests/cumulocity/remote-access/test_remote_access.robot:30]: ah ok. Then ignore my comment :)

**Pull request merged**